control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -261,7 +261,7 @@ func (r *HostedControlPlaneReconciler) registerComponents() {
 		snapshotcontrollerv2.NewComponent(),
 	)
 	r.components = append(r.components,
-		olmv2.NewComponents(r.ManagementClusterCapabilities.Has(capabilities.CapabilityImageStream))...,
+		olmv2.NewComponents(r.ImageMetadataProvider, r.ManagementClusterCapabilities.Has(capabilities.CapabilityImageStream))...,
```

Comment:
ImageMetadataProvider is already passed down in the cpContext, no need to pass it here.